pkg/agent/openflow/pipeline.go

```diff
@@ -456,9 +502,10 @@ func (c *client) gatewayIPSpoofGuardFlow(gatewayOFPort uint32, category cookie.C
 }
 
 // serviceCIDRDNATFlow generates flows to match dst IP in service CIDR and output to host gateway interface directly.
-func (c *client) serviceCIDRDNATFlow(serviceCIDR *net.IPNet, gatewayOFPort uint32, category cookie.Category) binding.Flow {
+func (c *client) serviceCIDRDNATFlow(serviceCIDR *net.IPNet, gatewayMAC net.HardwareAddr, gatewayOFPort uint32, category cookie.Category) binding.Flow {
 	return c.pipeline[dnatTable].BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
 		MatchDstIPNet(*serviceCIDR).
+		Action().SetDstMAC(gatewayMAC).
```

Comment:
Sorry, but why need this action now?

Reply:
unlike in encap case where we know default route is pointing to gw0. In policy-only mode, antrea has no idea what Pod's route table looks like, and it is next hop and Mac, Thus this flow used in conjunction with arpResponderStaticFlow steers service traffic to gw0.

Reply:
Why not return gw0 MAC in arpResponderStaticFlow?

Reply:
@tnqn @wenyingd: remind me why we do not use the virtual MAC for gw0?

Reply:
I think in encap mode, pod default route nh is to gw0, thus gw0 would response with its correct Mac address, thus no need to use virtual MAC

Reply:
we can return gw0 MAC in arpResponderStaticFlow.

Reply:
@jianjuns the arp responder is only installed for remote ips to prevent arp broadcast off the node, we didn't have any arp responder for local ips, regardless of pods or gw0, all of them learn the MAC by getting the real ARP reply. I think it's just we didn't see a good reason to do extra work when the traditional way can work.

Reply:
@tnqn : my question is if we set gw0 to use virtual MAC (it is also good to reduce chances of MAC conflicts), could our flows be simpler (like to avoid rewrite MAC for remote traffic).

I just do not like to have too many MACs - gw0 MAC, GW virtual MAC, and Su's new virtual MAC. Hard to understand which is for what.

But probably this is irrelevant to this PR.

Reply:
@jianjuns There could be some side effect to make gw0 use the same virtual MAC, the same MAC can appear on tunnel port and gw0 port, which can confuse ovs fdb table, not sure whether we rely on it or not right now, but there was forwarding issue because of it. And when gw0 talks to other nodes, the packets it sent will have same src and dst mac, not sure if it could lead to any issue.
I think currently the vmac is not complex, the vmac just represents the other side of tunnel, both from ingress and egress's perspective, all local macs are real and learnt from ARP. If we share vmac between gw0 and remote endpoints, it sounds a little confusing that which this packet targets for from L2 layer.

Back to this new vmac, per today's discussion, since the packets sent back to OVS for policy enforcement lost its original source IP and ofport, there's no way for policy to take effect, is the mac rewriting still needed?